src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -3014,6 +3014,30 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
         }
     }
 
+    void visit_StructType(const ASR::StructType_t &x){
+        SymbolTable *current_scope_copy = current_scope;
+        current_scope = x.m_symtab;
+        for (auto &item : x.m_symtab->get_scope()){
+            if (is_a<ASR::Function_t>(*item.second)){
+                ASR::Function_t *v = down_cast<ASR::Function_t>(item.second);
+                instantiate_function(*v);
+            }
+        }
+        current_scope = current_scope_copy;
+    }
+
+    void make_struct_f_def(const ASR::StructType_t &x){
```

Comment:
If you have time, try simplifying this function: visit_StructType and make_struct_f_def.
They seem duplicated